src/zenml/pipelines/base_pipeline.py

```diff
@@ -158,9 +159,18 @@ def _verify_arguments(self, *steps: BaseStep, **kw_steps: BaseStep) -> None:
                     f"a pipeline."
                 )
 
+            step_ref_func = type(step)
```

Comment:
I think `step_class` is clearer and covers both the functional and class-based API. Maybe we can even define it right after the for-loop and use it inside the previous error message (which also calls `type(step)`?

Reply:
yup sorry I will change this as well. I couldn't think of a good variable name initially.